promql/engine.go

```diff
@@ -1278,12 +1279,12 @@ func (ev *evaluator) rangeEval(prepSeries func(labels.Labels, *EvalSeriesHelper)
 			}
 			if sample.H == nil {
 				if ss.Floats == nil {
-					ss.Floats = getFPointSlice(numSteps)
+					ss.Floats = getFPointSlice(pointsSliceSize(numSteps, step, ev.currentSamples, ev.maxSamples, biggestLen))
```

Comment:
Seems ugly repeating this six times.  Could the logic move inside `getFPointSlice`, and make it a member of `evaluator` so it has current and max samples available?

Reply:
I will do that but there is some cases where we fix the `len` to 16.
I will do the change and we can see if its too ugly! hahah 
https://github.com/prometheus/prometheus/blob/8ef7dfdeebf0a7491973303c7fb6b68ec5cc065b/promql/engine.go#L2089